### PR TITLE
Update blog post. 

### PR DESCRIPTION
**Pull request by shaojingli**
Update screenshots. 
Change the language. 
Make the title left-aligned.
Add more links.

### CODE REVIEW

**Comment by shaojingli** [comment]
@Pessimistress

File: _posts/2017-03-31-introducing-deckgl-v4.md [hunk 1]
```diff
@@ -9,21 +9,21 @@ tags:
...
 
-deck.gl v3 saw rapid adoption across a number of internal data visualization applications here at Uber, and it didn't take long before an impressive list of new layers had been developed to support these applications. The most general of these layers soon started to be reused between applications, and it quickly became clear that including the "best" of these layers in a new official deck.gl release would likely be of high value both to internal and external developers.
+deck.gl v3 saw rapid adoption across a number of internal data visualization applications here at Uber, and it didn't take long before an impressive list of new layers had been developed to support these applications. For layers are reused between multiple applications, it became clear that open sourcing them in a new official deck.gl release would likely to benefit both to internal and external developers.
```
Comment (ibgreen) [inline _posts/2017-03-31-introducing-deckgl-v4.md]: > For layers ^ are reused between multiple applications

Missing `that` (layers that are ...)

Maybe say "for layers that `were`" to match the tense of the previous sentences?

Comment (ibgreen) [inline _posts/2017-03-31-introducing-deckgl-v4.md]: > release would ^ likely to benefit both to internal and external developers.

missing `be` (would be likely).

File: _posts/2017-03-31-introducing-deckgl-v4.md [hunk 2]
```diff
@@ -9,21 +9,21 @@ tags:
...
 
-In deck.gl v4, the core layer catalog in deck.gl effectively doubles in size to over a dozen carefully audited, tested and documented [layers](https://uber.github.io/deck.gl/#/documentation/layer-catalog), covering a significantly wider array of geospatial visualization use cases.
+In deck.gl v4, the core layer catalog in deck.gl effectively doubles in size to over a dozen. They are carefully audited, tested and documented [layers](https://uber.github.io/deck.gl/#/documentation/layer-catalog), covering a significantly wider array of geospatial visualization use cases.
```
Comment (ibgreen) [inline _posts/2017-03-31-introducing-deckgl-v4.md]: > doubles in size to over a dozen ^.

This sentence sounds cut off. "...to over a dozen `layers`."


> They ^ are carefully audited, tested and documented [layers]

Would suggest: `These layers have been carefully audited, ...`

File: _posts/2017-03-31-introducing-deckgl-v4.md [hunk 3]
```diff
@@ -33,75 +33,74 @@ is provided and we expect the upgrade effort for applications to be minimal.
 
 ## A Growing Layer Catalog
 
-From the very beginning, deck.gl has always been a geospatial data visualization framework, and geospatial use cases will always be the top priority. The new layers added to the [layer catalog](https://uber.github.io/deck.gl/#/documentation/layer-catalog) are widely used in geospatial data analysis within Uber and we hope they will help solve the problems that deck.gl users face every day.
+From the very beginning, deck.gl has always been a geospatial data visualization framework, and geospatial use cases will always be our top priority. The new layers added to the [layer catalog](https://uber.github.io/deck.gl/#/documentation/layer-catalog) are widely used in geospatial data analysis within Uber and we hope they will help solve the problems that every deck.gl user faces every day.
```
Comment (ibgreen) [inline _posts/2017-03-31-introducing-deckgl-v4.md]: > that ^every deck.gl user faces ^every day

Repeating a word (`every`) in a sentence, avoid if possible.

File: _posts/2017-03-31-introducing-deckgl-v4.md [hunk 4]
```diff
@@ -33,75 +33,74 @@ is provided and we expect the upgrade effort for applications to be minimal.
...
 ### [`GeoJsonLayer`](https://uber.github.io/deck.gl/#/documentation/layer-catalog/geojson-layer)
 
-<p>
-	<img width="400" src="../img/demo-thumb-geojson.jpg">
+<p aligh="center">
+	<img width="900" src="../img/screenshot-geojson.png">
 </p>
 
-The GeoJsonLayer is designed to render geometry primitives in the widely accepted GeoJson data format. The GeoJsonLayer itself is very powerful as it renders multiple types of geometry primitives, including points, lines, multilines, polygons and multipolygons all at once. Moreover, it can render polygons and multipolygons in different styles, currently supporting filled and outline styles in 2D, and filled and wireframe styles in extruded.
+The GeoJsonLayer is designed to render geometry primitives in the widely accepted GeoJson data format. The GeoJsonLayer itself is very powerful as it renders multiple types of geometry primitives, including points, lines, multilines, polygons and multipolygons all at once. Moreover, it can render polygons and multipolygons in different styles, currently supporting filled and outline styles in 2D flat, and filled and wireframe styles in extruded.
```
Comment (ibgreen) [inline _posts/2017-03-31-introducing-deckgl-v4.md]: > currently supporting filled and outline styles in 2D flat, and filled and wireframe styles in extruded

Yes this needed fixing. "2D flat" is still an odd term. Maybe even clearer:

`currently supporting filled and outline styles in non-extruded (2D) mode, and filled and wireframe styles in extruded (3D) mode.`

File: _posts/2017-03-31-introducing-deckgl-v4.md [hunk 5]
```diff
@@ -33,75 +33,74 @@ is provided and we expect the upgrade effort for applications to be minimal.
...
 
 ### [`PolygonLayer`](https://uber.github.io/deck.gl/#/documentation/layer-catalog/polygon-layer)
 
 <p aligh="center">
-	<img width="400" src="../img/demo-thumb-polygon.png">
+	<img width="900" src="../img/screenshot-polygon.png">
 </p>
 
-The `PolygonLayer` renders general flat or extruded polygons specified by a "closed" sequence of coordinates, like how polygon features are specified in the GeoJson specifications. Polygons can be rendered as filled or outline in 2D and filled or wireframe in 3D. This layer supports rendering convex, concave and polygon with holes.
+The `PolygonLayer` renders general flat or extruded polygons specified by a "closed" sequence of coordinates, like how polygon features are specified in the GeoJson specifications. Polygons can be rendered as filled or outline in 2D flat and filled or wireframe in extruded. This layer supports rendering convex, concave and polygon with holes.
```
Comment (ibgreen) [inline _posts/2017-03-31-introducing-deckgl-v4.md]: As above, if changing there let's change here too

File: _posts/2017-03-31-introducing-deckgl-v4.md [hunk 6]
```diff
@@ -33,75 +33,74 @@ is provided and we expect the upgrade effort for applications to be minimal.
...
 
 Aggregation is an exiciting new direction for deck.gl as we want to not only provide layers that are visually compelling to our users, but are also ***smart***.
 
-Finally, note that the new `GridLayer` and `HexagonLayer` are different from the existing `ScreenGridLayer` as both the rendered geometry and the aggregation boundaries are in world coordinates rather than screen coordinates. Another difference is that the new layers also support both 2D and 3D cells.
+Finally, note that the new `GridLayer` and `HexagonLayer` are different from the existing `ScreenGridLayer` as both the rendered geometry and the aggregation boundaries are in world coordinates rather than screen coordinates. Another difference is that the new layers also support both 2D flat and 3D extruded cells.
```
Comment (ibgreen) [inline _posts/2017-03-31-introducing-deckgl-v4.md]: > both 2D flat and 3D extruded cells

`flat (2D) and extruded (3D) cells`?

File: _posts/2017-03-31-introducing-deckgl-v4.md [hunk 7]
```diff
@@ -33,75 +33,74 @@ is provided and we expect the upgrade effort for applications to be minimal.
...
 
 ## Layer extrusion and lighting
 
-Some of deck.gl layers now support extrusion. Extrusion can be enabled by setting the `extruded` property to true, if available. Users can provide a `getElevation` accessor to control how much each individual element would extrude.
+As mentioned in previous sections, some of deck.gl layers now support extrusion. Extrusion can be enabled by setting the `extruded` property to true, if available. Users can provide a `getElevation` accessor to control how much each individual element would extrude.
```
Comment (ibgreen) [inline _posts/2017-03-31-introducing-deckgl-v4.md]: > some of ^ deck.gl layers 
Missing `the`

Comment (ibgreen) [inline _posts/2017-03-31-introducing-deckgl-v4.md]: > how much each individual element would extrude.

`would` => `should`

File: _posts/2017-03-31-introducing-deckgl-v4.md [hunk 8]
```diff
@@ -33,75 +33,74 @@ is provided and we expect the upgrade effort for applications to be minimal.
...
 
-To make extrusion looks natural and make extruded layers look visually pleasing, deck.gl v4 includes an experimental lighting module that provides basic shading of 3D layers. It has very straightforward APIs and should be easy for users to understand by simplying looking at every increasing number of examples that comes with deck.gl. It's worth mentioning that since it's experimental, the public APIs might change in the future release of deck.gl.
+To make extrusion looks natural and visually pleasing, deck.gl v4 now has an experimental lighting module that provides basic shading of 3D layers. It has very straightforward APIs and it should be easy for users to understand by simplying looking at every increasing number of examples that comes with deck.gl. It's worth mentioning that since it's experimental, the public APIs might change in the future release of deck.gl.
```
Comment (ibgreen) [inline _posts/2017-03-31-introducing-deckgl-v4.md]: > by simplying looking at every increasing number of examples that comes with deck.gl

Change to: `simply by looking at the examples that come with deck.gl`

File: _posts/2017-03-31-introducing-deckgl-v4.md [hunk 9]
```diff
@@ -110,35 +109,31 @@ deck.gl v3 came with essentially a single example that was interleaved with the
...
 
-As we march forward in the geospatial data analysis field, it's obvious that the single threaded Javascript engine in our browsers will soon be overwhelmed by computation intensive tasks. We can solve this by seeking help from backend, but not without its own limitations. The recently released WebGL 2.0 provided a way to access powerful GPUs on every desktops and laptops in a way that is not possible in the past. With features like transform feedback, we might be able to vastly accelerate many computational tasks in geospatial data analysis.
+As we march forward in the geospatial data analysis field, it's obvious that the single threaded Javascript engine in our browsers will soon be overwhelmed by computation intensive tasks. We can solve this by seeking help from backend, but not without its own limitations. The recently released WebGL 2.0 provides a way for front-end developers to access powerful GPUs on every desktops and laptops that is not possible in the past. With features like [transform feedback](https://developer.mozilla.org/en-US/docs/Web/API/WebGL2RenderingContext), we might be able to vastly accelerate many computational tasks in geospatial data analysis.
```
Comment (ibgreen) [inline _posts/2017-03-31-introducing-deckgl-v4.md]: > We can solve this by seeking help from backend

Missing the

`from the  backend`, or `from backend services`

Comment (ibgreen) [inline _posts/2017-03-31-introducing-deckgl-v4.md]: > The recently released WebGL 2.0 `standard` provides

**Comment by Pessimistress** [comment]
Can we not show the tooltips front and center on every screenshot? They're not very good looking.

**Comment by shaojingli** [comment]
@Pessimistress screenshot changed.
@ibgreen thanks a lot for going words by words. updated according to comments

File: _posts/2017-03-31-introducing-deckgl-v4.md [hunk 10]
```diff
@@ -9,21 +9,21 @@ tags:
...
+deck.gl v3 saw rapid adoption across a number of internal data visualization applications here at Uber, and it didn't take long before an impressive list of new layers had been developed to support these applications. For layers that were reused between multiple applications, it became clear that open sourcing them in a new official deck.gl release would be likely to benefit both to internal and external developers.
 
-In deck.gl v4, the core layer catalog in deck.gl effectively doubles in size to over a dozen carefully audited, tested and documented [layers](https://uber.github.io/deck.gl/#/documentation/layer-catalog), covering a significantly wider array of geospatial visualization use cases.
+In deck.gl v4, the core layer catalog in deck.gl effectively doubles in size to over a dozen [layers](https://uber.github.io/deck.gl/#/documentation/layer-catalog). These layers are carefully audited, tested and documented, covering a significantly wider array of geospatial visualization use cases.
```
Comment (ibgreen) [inline _posts/2017-03-31-introducing-deckgl-v4.md]: `covering a` => `and cover`

File: _posts/2017-03-31-introducing-deckgl-v4.md [hunk 11]
```diff
@@ -33,75 +33,74 @@ is provided and we expect the upgrade effort for applications to be minimal.
...
 
 ### [`PolygonLayer`](https://uber.github.io/deck.gl/#/documentation/layer-catalog/polygon-layer)
 
 <p aligh="center">
-	<img width="400" src="../img/demo-thumb-polygon.png">
+	<img width="900" src="../img/screenshot-polygon.png">
 </p>
 
-The `PolygonLayer` renders general flat or extruded polygons specified by a "closed" sequence of coordinates, like how polygon features are specified in the GeoJson specifications. Polygons can be rendered as filled or outline in 2D and filled or wireframe in 3D. This layer supports rendering convex, concave and polygon with holes.
+The `PolygonLayer` renders general flat or extruded polygons specified by a "closed" sequence of coordinates, like how polygon features are specified in the GeoJson specifications. Polygons can be rendered as filled or outline in non-extruded (2D) modet and filled or wireframe in extruded (3D) mode. This layer supports rendering convex, concave and polygon with holes.
```
Comment (ibgreen) [inline _posts/2017-03-31-introducing-deckgl-v4.md]: Typo: `(2D) modet` => `(2D) mode`

File: _posts/2017-03-31-introducing-deckgl-v4.md [hunk 12]
```diff
@@ -33,75 +33,74 @@ is provided and we expect the upgrade effort for applications to be minimal.
...
 
-Finally, note that the new `GridLayer` and `HexagonLayer` are different from the existing `ScreenGridLayer` as both the rendered geometry and the aggregation boundaries are in world coordinates rather than screen coordinates. Another difference is that the new layers also support both 2D and 3D cells.
+Finally, note that the new `GridLayer` and `HexagonLayer` are different from the existing `ScreenGridLayer` as both the rendered geometry and the aggregation boundaries are in world coordinates rather than screen coordinates. Another difference is that the new layers also support both flat (2D) and extruded (3D) cells.
 
 ## Layer extrusion and lighting
 
-Some of deck.gl layers now support extrusion. Extrusion can be enabled by setting the `extruded` property to true, if available. Users can provide a `getElevation` accessor to control how much each individual element would extrude.
+As mentioned in the previous sections, some of deck.gl layers now support extrusion. Extrusion can be enabled by setting the `extruded` property to true, if available. Users can provide a `getElevation` accessor to control how much each individual element should extrude.
```
Comment (ibgreen) [inline _posts/2017-03-31-introducing-deckgl-v4.md]: `some of deck.gl layers` => `some deck.gl layers`

File: _posts/2017-03-31-introducing-deckgl-v4.md [hunk 13]
```diff
@@ -33,75 +33,74 @@ is provided and we expect the upgrade effort for applications to be minimal.
...
 
-To make extrusion looks natural and make extruded layers look visually pleasing, deck.gl v4 includes an experimental lighting module that provides basic shading of 3D layers. It has very straightforward APIs and should be easy for users to understand by simplying looking at every increasing number of examples that comes with deck.gl. It's worth mentioning that since it's experimental, the public APIs might change in the future release of deck.gl.
+To make extrusion looks natural and visually pleasing, deck.gl v4 now has an experimental lighting module that provides basic shading of 3D layers. It has very straightforward APIs and it should be easy for users to understand simply by looking at the examples that come with deck.gl. It's worth mentioning that since it's experimental, the public APIs might change in the future release of deck.gl.
```
Comment (ibgreen) [inline _posts/2017-03-31-introducing-deckgl-v4.md]: `To make extrusion looks natural` => `To make extrusion look natural` or `To make extrusions looks natural`

`It has very straightforward APIs` => `It has a straightforward API`

File: _posts/2017-03-31-introducing-deckgl-v4.md [hunk 14]
```diff
@@ -110,35 +109,31 @@ deck.gl v3 came with essentially a single example that was interleaved with the
...
-deck.gl v4 is an important milestone for us, but what's next? While we haven't finalized
-any plans for a possible deck.gl v4.1 or v5 release yet, we have many ideas about
-improvements and directions we would like deck.gl to evolve along. Some examples are:
+deck.gl v4 is an important milestone for us, but what's next? While we haven't finalized any plans for a possible deck.gl v4.1 or v5 release yet, we have many ideas about improvements and directions we would like deck.gl to evolve along. Some examples are:
 
 ### More and Better Layers
 
-Layers are how deck.gl creates value for most users, so obviously we plan to add more layers. We will of course always keep improving support for the geospatial vusalizations but we are also
-considering supporting abstract visualizations in the InfoVis area.
+Layers are how deck.gl creates value for most users, so adding more layers is on the top of our list. We will always keep improving support for the geospatial vusalizations while also considering supporting abstract visualizations in the InfoVis area.
```
Comment (ibgreen) [inline _posts/2017-03-31-introducing-deckgl-v4.md]: Is `InfoVis` an obvious term, especially with this distinctive capitalization? 

As it is suddenly mentioned, should there be a link to wikipedia or similar? 

@gnavvy @ericsoco

Comment (ericsoco) [inline _posts/2017-03-31-introducing-deckgl-v4.md]: "non-geospatial" is probably more descriptive.

**Pull request merged**